### PR TITLE
Retry requests

### PR DESCRIPTION
**Pull request by raychenon**
Retry failed requests in case of network / remote server failures.
https://github.com/zalando-incubator/play-zhewbacca/issues/16

**Note :** The number of attempts and backoff duration variables need to be extracted in .conf file .

### CODE REVIEW

File: src/main/scala/org/zalando/zhewbacca/IAMClient.scala [hunk 1]
```diff
@@ -85,4 +92,10 @@ class IAMClient @Inject() (
       }
   }
 
+  implicit val retryRecover = retryFor { 3.attempts } using {
+    exponentialBackoff { 100.millis }
```
Comment (raychenon) [inline src/main/scala/org/zalando/zhewbacca/IAMClient.scala]: these variables ( nb attempts and backoff duration ) should be configurable in `.conf` file

**Comment by lasomethingsomething** [comment]
@raychenon Are you guys planning on merging this PR after resolving the conflict?

**Comment by raychenon** [comment]
@LappleApple , thanks for the reminder.
Conflict resolved.
We plan to merge this PR this week.

File: src/main/scala/org/zalando/zhewbacca/IAMClient.scala (hunk 1 above)
Comment (dmitrykrivaltsevich) [reply src/main/scala/org/zalando/zhewbacca/IAMClient.scala]: Could you please make then configurable?

**Review by dmitrykrivaltsevich — CHANGES_REQUESTED** [review]

File: src/main/scala/org/zalando/zhewbacca/IAMClient.scala (hunk 1 above)
Comment (raychenon) [reply src/main/scala/org/zalando/zhewbacca/IAMClient.scala]: I'll do it today. Is there anything else to review ?

Comment (dmitrykrivaltsevich) [reply src/main/scala/org/zalando/zhewbacca/IAMClient.scala]: Nope

**Review by dmitrykrivaltsevich — APPROVED** [review]

**Comment by dmitrykrivaltsevich** [comment]
@raychenon , could you please release 0.2.2?

**Comment by raychenon** [comment]
@dmitrykrivaltsevich , I will release tomorrow . My machine is at work.

**Comment by dmitrykrivaltsevich** [comment]
- resolves #16

**Comment by dmitrykrivaltsevich** [comment]
@raychenon , was 0.2.2 already released?

**Pull request merged**